buildSrc/src/main/groovy/io/micronaut/build/internal/ext/DefaultMicronautCoreExtension.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2003-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.build.internal.ext;
+
+import org.gradle.api.artifacts.ExternalModuleDependency;
+import org.gradle.api.artifacts.VersionCatalog;
+import org.gradle.api.artifacts.VersionCatalogsExtension;
+import org.gradle.api.artifacts.dsl.DependencyHandler;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.Map;
+
+public abstract class DefaultMicronautCoreExtension implements MicronautCoreExtension {
+    private final DependencyHandler dependencyHandler;
+    private final VersionCatalog libs;
+
+    @Inject
+    public DefaultMicronautCoreExtension(DependencyHandler dependencyHandler, VersionCatalogsExtension versionCatalogs) {
+        this.dependencyHandler = dependencyHandler;
+        this.libs = versionCatalogs.find("libs").get();
+    }
+
+    private static void excludeMicronautLibs(ExternalModuleDependency dep) {
+        dep.exclude(module("micronaut-runtime"));
+        dep.exclude(module("micronaut-inject"));
+        dep.exclude(module("micronaut-bom"));
```

Comment:
In this PR, you define five modules as core libraries `inject`, `aop`, `core`, `runtime`, `session`.

In addtion to `bom`, this excludes only `runtime` and `inject` but not `core`, `aop` or `session`. Is this correct?

Reply:
That's correct. Again I'm not sure if this is intentional or not: this was what was done before. Note that I have checked that the produced jars, pom and module files are identical _byte-to-byte_ using this PR.

Reply:
> The build won't pass: after upgrading to build tools 5.3.0 it now detects binary breaking changes:

Why would it fail? 

Should the binary check not check if this PR introduces any breaking changes from 3.4.0 to this PR?

Reply:
It's not failing because of changes in this PR. It's failing because it upgrades to build plugins 5.3.0, which introduce binary compatibility checks, which fail because they detect binary incompatible changes between 3.4.1-SNAPSHOT and 3.4.0 (see screenshots in comments).

Reply:
yes, but those changes have not been introduced between 3.4.0 and 3.4.1-SNAPSHOT.  It seems the binary check is comparing 3.3.4 and 3.4.0 not 3.4.0 and 3.4.1-SNAPSHOT

Reply:
No I think the problem was that it wasn't up-to-date with `3.4.x`. The binary checks pass now. However now it's BOM checking which doesn't :D

Reply:
If I understand correctly.  We have to fix: 
- [ ] micronaut-r2dbc-bom
- [ ] micronaut-servlet-bom 
- [ ] micronaut-oraclecloud-bom
- [ ] micronaut-data 
- [ ]micronaut-serde-bom

It will involve releasing versions of those modules updating the core version catalogue with those new versions. Then merge 3.4.x branch against this PR again.  

right?

Reply:
If Updating those modules to 5.3.0, it should trigger these errors in their builds. right?

Reply:
There's nothing we can do for r2dbc (apart from silencing the problem) as far as I understood, unless we want to inline the dependencies ourselves.
For the other BOMs I have created issues in the projects and silenced the problems here so that we can proceed _without_ waiting for new releases.
Yes, migrating those projects to 5.3.0 would trigger errors in their own builds.